src/sortStudents.ts

```diff
@@ -1,16 +1,93 @@
 
 export interface Student {
   // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
   // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'avg',
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
-
-export function sortStudents(students, sortBy, order) {
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
   // write your function
+  const clonedStudents = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      if (order === 'asc') {
+        clonedStudents.sort(
+          (student, student2) => student[sortBy]
+            .localeCompare(student2[sortBy]),
+        );
+        break;
+      }
+
+      clonedStudents.sort(
+        (student, student2) => student2[sortBy]
+          .localeCompare(student[sortBy]),
+      );
+      break;
+
+    case SortType.Age:
+    case SortType.Married:
+      if (order === 'asc') {
+        clonedStudents.sort(
+          (student, student2) => +student[sortBy] - +student2[sortBy],
+        );
+        break;
+      }
+
+      clonedStudents.sort(
+        (student, student2) => +student2[sortBy] - +student[sortBy],
+      );
```

Comment:
It probably would look something like this with ternary operators. Try it out and if you have any questions about how it works, you can come to the QnA. It might be a black magic, but they are very usefull
```suggestion
      return order === 'asc'
        ?  clonedStudents.sort(
          (student, student2) => +student[sortBy] - +student2[sortBy],
        );
        : clonedStudents.sort(
        (student, student2) => +student2[sortBy] - +student[sortBy],
      );
```

Reply:
Doing this causes the sort to not work

Reply:
..Nevermind, I was looking at the wrong line